From 8fd7162268038c505ca9b7508708d2c7ad9db94b Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Wed, 14 Nov 2018 15:01:30 +0200
Subject: [PATCH 5139/5242] MLK-20216-5: video/fbdev/mxc_dcic: remove needless
 comparison

commit  2036a2a4ead995f0dfce999b68304b33806b390b from
https://source.codeaurora.org/external/imx/linux-imx.git

This fixes less-then-zero comparison Coverity issue. roi_param->roi_n is
unsigned, hence the comparison is not needed.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
CC: Sandor Yu <sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mxc_dcic.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxc/mxc_dcic.c b/drivers/video/fbdev/mxc/mxc_dcic.c
index cae872a..3c5a83be 100644
--- a/drivers/video/fbdev/mxc/mxc_dcic.c
+++ b/drivers/video/fbdev/mxc/mxc_dcic.c
@@ -230,7 +230,7 @@ static bool roi_configure(struct dcic_data *dcic, struct roi_params *roi_param)
 	struct roi_regs *roi_reg;
 	u32 val;
 
-	if (roi_param->roi_n < 0 || roi_param->roi_n >= 16) {
+	if (roi_param->roi_n >= 16) {
 		printk(KERN_ERR "Error, Wrong ROI number %d\n", roi_param->roi_n);
 		return false;
 	}
-- 
1.7.9.5

